python/cudf/cudf/core/series.py

```diff
@@ -4617,8 +4617,73 @@ def _get_dt_field(self, field):
             data=out_column, index=self.series._index, name=self.series.name
         )
 
-    def ceil(self, field):
-        out_column = self.series._column.ceil(field)
+    def ceil(self, freq):
+        """
+        Perform ceil operation on the data to the specified freq.
+
+        Parameters
+        ----------
+        freq : str
+            One of ["D", "H", "T", "S", "L", "U", "N"]
```

Comment:
May need to adapt to accept `"min"`, `"ms"`, `"us"`, `"ns"` for the aliases listed here: https://pandas.pydata.org/docs/user_guide/timeseries.html#timeseries-offset-aliases

Reply:
@bdice added 👍, I wanted to do something like `in` in the changes I posted, eg:

```python
if field in ("L", "ms")
```

but I think `cython` is still working to support that operation (https://github.com/cython/cython/issues/3652)

Reply:
@mayankanand007 Can you make the same changes for `ceil`? Generally any improvements you've made here to `floor` for implementation/testing/etc. should also apply to `ceil`.

The issue with `field in ("L", "ms")` is that you can't perform the `in` operation when the GIL has been released. (If you're not familiar with the GIL, you can [get a brief overview here](https://docs.python.org/3/c-api/init.html#thread-state-and-the-global-interpreter-lock). It's a complex topic.) The tuple `("L", "ms")` is a Python object, not a C++ Standard Library (STL) container, so this is a GIL problem and not the issue you linked with checking containment in an STL container.

Reply:
@mayankanand007 Also you might have missed an alias for `ns` -- nanoseconds in 53b7b3b37a6c47b33f2bd524f8c99c3c9a36d759.

Reply:
Thanks @bdice for clarifying, as for adding `ns`, I couldn't find it in the list here: https://pandas.pydata.org/docs/user_guide/timeseries.html#timeseries-offset-aliases next to `nanoseconds`, which is why I didn't add it.

Reply:
Sorry, I deleted that comment because I questioned myself after posting. You're correct, there's not an alias for `ns`.

Reply:
no problem, I couldn't find it here on GiHub, but got a notification for it, which confused me a little, all good 😄